src/Autumn/compile.jl

```diff
@@ -2,24 +2,306 @@
 module Compile
 
 using ..AExpressions
-using ..Program
+using Distributions: Categorical
+using MLStyle: @match
+import MacroTools: striplines
 
-"Compile `aexpr` into `program::Program`"
-function compiletojulia(::AExpr)::AProgram
-  # Do type inference
-  # Get external values
-  # Alot
+export compiletojulia, runprogram
+
+binaryOperators = [:+, :-, :/, :*, :&, :|, :>=, :<=, :>, :<, :(==), :!=, :%, :&&]
+
+struct AutumnCompileError <: Exception
+  msg
+end
+AutumnCompileError() = AutumnCompileError("")
+
+"compile `aexpr` into Expr"
+function compiletojulia(aexpr::AExpr)::Expr
+
+  data = Dict([("historyVars" => []),
+               ("externalVars" => []),
+               ("initnextVars" => []),
+               ("liftedVars" => []),
+               ("types" => Dict())])
+  
+  # ----- HELPER FUNCTIONS ----- #
+  function compile(expr::AExpr, parent=nothing)
```

Comment:
Avoid having this huge function, take all these helper functions out and pass `data` through (which you are already doing in most cases anyway).  Currently they cannot be tested in isolation and the big function is unwieldy

Reply:
Also once you have taken out these helper functions add a tests for each

For instance

`@test compileassign(AExpr(....)) = :(const x = 3)`